### PR TITLE
Fix webhook subscription creation for SharePoint Libraries

### PR DESCRIPTION
**Pull request by abraunegg**
* Create a resource item using the driveId rather than specifying /me/drive/root

### CODE REVIEW

File: src/onedrive.d [hunk 1]
```diff
@@ -1012,10 +1012,12 @@ final class OneDriveApi
 		auto expirationDateTime = Clock.currTime(UTC()) + subscriptionExpirationInterval;
 		const(char)[] url;
 		url = subscriptionUrl;
+		// Create a resource item using the account driveId rather than using /me/drive/root
```
Comment (MouettE-SC) [inline src/onedrive.d:1015]: This only works when driveId is set in the configuration. I suggest this code which works well for both personal, business and sharepoint drives :

```
string resourceItem;
if (driveId.length) {
        resourceItem = "/drives/" ~ driveId ~ "/root";
} else {
        resourceItem = "/me/drive/root";
}

```

**Review by MouettE-SC — CHANGES_REQUESTED** [review]

File: src/onedrive.d (hunk 1 above)
Comment (abraunegg) [reply src/onedrive.d:1015]: This would have been the updated PR once feedback from you. I will update the PR with the above.

**Review by MouettE-SC — APPROVED** [review]
This fixes the issue for Sharepoint drives and does not seem to introduce regression in the webhook functionality.

**Comment by abraunegg** [comment]
This pull request has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**